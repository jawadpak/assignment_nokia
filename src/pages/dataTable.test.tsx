import React from "react";
import {
  render,
  screen,
  RenderResult,
  waitFor,
  queryByTestId,
  waitForElement,
  cleanup,
  findByTestId,
  fireEvent
} from "@testing-library/react";
import { shallow } from "enzyme";
import Enzyme, { mount } from "enzyme";
import { Checkbox, Switch } from "@material-ui/core";

import DataTable from "./dataTable";

import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

describe("<ToggleAutoRefresh />", () => {
  it("renders with Props", async () => {
    const wrapper = render(<DataTable />);
    expect(wrapper).toMatchSnapshot();
  });
  //   it("handels click", async () => {
  //     const handleChangeAutoRefresh = jest.fn();
  //     const wrapper = render(
  //       <ToggleAutoRefresh
  //         autoRefresh={true}
  //         handleChangeAutoRefresh={handleChangeAutoRefresh}
  //       />
  //     );
  //     const switchBtn = await wrapper.findAllByTestId("rdAutoRefresh");
  //     switchBtn[0].click();
  //     expect(handleChangeAutoRefresh).toBeCalledTimes(1);
  //  });
});
