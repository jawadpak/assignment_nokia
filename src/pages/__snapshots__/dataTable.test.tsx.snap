// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleAutoRefresh /> renders with Props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%;"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular makeStyles-root-3 MuiToolbar-gutters"
        >
          <div
            class="MuiTypography-root makeStyles-title-5 MuiTypography-h6"
            id="tableTitle"
          >
            Nokian liput
          </div>
          <button
            aria-label="filter list"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            title="Filter list"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            aria-label="enhanced table"
            class="MuiTable-root makeStyles-table-1"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  colspan="2"
                  scope="col"
                >
                  <label
                    class="MuiFormControlLabel-root"
                    data-testid="autoRefresh"
                  >
                    <span
                      class="MuiSwitch-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
                        data-testid="rdAutoRefresh"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            class="PrivateSwitchBase-input-9 MuiSwitch-input"
                            name="rdAutoRefresh"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="MuiSwitch-thumb"
                          />
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiSwitch-track"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Automaattinen p√§ivitys
                    </span>
                  </label>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  colspan="3"
                  scope="col"
                >
                  <div
                    aria-expanded="false"
                    class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                    role="combobox"
                    style="width: 300px;"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                        data-shrink="false"
                        for="combo-box-demo"
                        id="combo-box-demo-label"
                      >
                        Hae Auto Ticket
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-invalid="false"
                          autocapitalize="none"
                          autocomplete="off"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                          id="combo-box-demo"
                          spellcheck="false"
                          type="text"
                          value=""
                        />
                        <div
                          class="MuiAutocomplete-endAdornment"
                        >
                          <button
                            aria-label="Clear"
                            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                            tabindex="-1"
                            title="Clear"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                          <button
                            aria-label="Open"
                            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                            tabindex="-1"
                            title="Open"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="PrivateNotchedOutline-legendLabelled-12"
                          >
                            <span>
                              Hae Auto Ticket
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </th>
              </tr>
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  colspan="5"
                  scope="col"
                >
                  <nav
                    aria-label="pagination navigation"
                    class="MuiPagination-root"
                  >
                    <ul
                      class="MuiPagination-ul"
                    >
                      <li>
                        <button
                          aria-label="Go to previous page"
                          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                          disabled=""
                          tabindex="-1"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiPaginationItem-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                            />
                          </svg>
                        </button>
                      </li>
                      <li>
                        <button
                          aria-current="true"
                          aria-label="page 1"
                          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-selected"
                          tabindex="0"
                          type="button"
                        >
                          1
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </li>
                      <li>
                        <button
                          aria-label="Go to page 2"
                          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                          tabindex="0"
                          type="button"
                        >
                          2
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </li>
                      <li>
                        <button
                          aria-label="Go to page 3"
                          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                          tabindex="0"
                          type="button"
                        >
                          3
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </li>
                      <li>
                        <button
                          aria-label="Go to page 4"
                          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                          tabindex="0"
                          type="button"
                        >
                          4
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </li>
                      <li>
                        <button
                          aria-label="Go to next page"
                          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                          tabindex="0"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiPaginationItem-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </li>
                    </ul>
                  </nav>
                   
                </th>
              </tr>
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  colspan="5"
                  scope="col"
                >
                   
                  <div
                    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
                    role="progressbar"
                  >
                    <div
                      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
                    />
                    <div
                      class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
                    />
                  </div>
                </th>
              </tr>
              <tr
                class="MuiTableRow-root makeStyles-tblHeader-2 MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                   
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        aria-label="select all record"
                        class="PrivateSwitchBase-input-9"
                        data-indeterminate="false"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Operaatio
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Soveltamisala
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Aikaleima
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Tila
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-root-3 MuiToolbar-gutters"
      >
        <div
          class="MuiTypography-root makeStyles-title-5 MuiTypography-h6"
          id="tableTitle"
        >
          Nokian liput
        </div>
        <button
          aria-label="filter list"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          title="Filter list"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          aria-label="enhanced table"
          class="MuiTable-root makeStyles-table-1"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                colspan="2"
                scope="col"
              >
                <label
                  class="MuiFormControlLabel-root"
                  data-testid="autoRefresh"
                >
                  <span
                    class="MuiSwitch-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
                      data-testid="rdAutoRefresh"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-9 MuiSwitch-input"
                          name="rdAutoRefresh"
                          type="checkbox"
                          value=""
                        />
                        <span
                          class="MuiSwitch-thumb"
                        />
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Automaattinen p√§ivitys
                  </span>
                </label>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                colspan="3"
                scope="col"
              >
                <div
                  aria-expanded="false"
                  class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
                  role="combobox"
                  style="width: 300px;"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="combo-box-demo"
                      id="combo-box-demo-label"
                    >
                      Hae Auto Ticket
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                        id="combo-box-demo"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment"
                      >
                        <button
                          aria-label="Clear"
                          class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                          tabindex="-1"
                          title="Clear"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-label="Open"
                          class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                          tabindex="-1"
                          title="Open"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M7 10l5 5 5-5z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-12"
                        >
                          <span>
                            Hae Auto Ticket
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </th>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                colspan="5"
                scope="col"
              >
                <nav
                  aria-label="pagination navigation"
                  class="MuiPagination-root"
                >
                  <ul
                    class="MuiPagination-ul"
                  >
                    <li>
                      <button
                        aria-label="Go to previous page"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiPaginationItem-icon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                      </button>
                    </li>
                    <li>
                      <button
                        aria-current="true"
                        aria-label="page 1"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-selected"
                        tabindex="0"
                        type="button"
                      >
                        1
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 2"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                        tabindex="0"
                        type="button"
                      >
                        2
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 3"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                        tabindex="0"
                        type="button"
                      >
                        3
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to page 4"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                        tabindex="0"
                        type="button"
                      >
                        4
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </li>
                    <li>
                      <button
                        aria-label="Go to next page"
                        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiPaginationItem-icon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </li>
                  </ul>
                </nav>
                 
              </th>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                colspan="5"
                scope="col"
              >
                 
                <div
                  class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
                  role="progressbar"
                >
                  <div
                    class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
                  />
                  <div
                    class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
                  />
                </div>
              </th>
            </tr>
            <tr
              class="MuiTableRow-root makeStyles-tblHeader-2 MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                 
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      aria-label="select all record"
                      class="PrivateSwitchBase-input-9"
                      data-indeterminate="false"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Operaatio
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Soveltamisala
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Aikaleima
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                scope="col"
              >
                Tila
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
